js/index.js

```diff
@@ -1,7 +1,105 @@
+  theShortest = hacker2;
+  } else if (hacker2.length > hacker1.length) {
+  console.log(`It seems that the navigator has the longest name, it has ${hacker2.length} characters.`);
+  theShortest = hacker1;
+  } else {
+    console.log(`Wow, you both have equally long names, ${hacker1.length} characters!`);
+    theShortest = hacker2;
+  }
 
+// Iteration 3.1
+const hacker1UpperCase = hacker1.toUpperCase();
+let hacker1Output = "";
+for (let character of hacker1UpperCase) {
+  hacker1Output += character;
+  if (character !== hacker1UpperCase.length -1) {
+    hacker1Output += " ";
+  }
+}
 
-// Iteration 3: Loops
\ No newline at end of file
+// Iteration 3.2
+let hacker2Output = "";
+for (i=hacker2.length -1; i>=0; i--) {
+  hacker2Output += hacker2[i];
+}
+
+console.log(hacker1Output);
+console.log(hacker2Output);
+
+// Iteration 3.3                       
+for (let i=0; i<theShortest.length; i++) {
+  if (hacker1[i].charCodeAt() < hacker2[i].charCodeAt()) {
+    console.log("The driver's name goes first.");
+    break;
+  } else if (hacker1[i].charCodeAt() > hacker2[i].charCodeAt()) {
+    console.log("The driver's name goes first.");
+    break;
+  } else if (hacker1 === hacker2) {
+    console.log("What?! You both have the same name?");
+    break;
+  }
+}
+
+// Bonus 1
+const paragraph = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam a porta velit. Mauris quis luctus ex. Donec non lacus ac velit hendrerit tempus. Nunc felis nunc, facilisis quis quam in, venenatis tristique ipsum. Sed molestie hendrerit tortor elementum rutrum. Ut varius dolor ac fermentum vehicula. Maecenas quis magna lobortis lectus ornare faucibus. Proin et fermentum ipsum. In vel auctor felis, eu malesuada elit. Donec vitae erat at quam fringilla fermentum. Aliquam erat volutpat. Morbi non cursus turpis. Suspendisse rutrum sodales justo tincidunt vehicula.\nCras nec molestie sem, et condimentum massa. Aenean ac elit mi. Praesent aliquet venenatis odio, a finibus tellus dictum a. Pellentesque vulputate nisl in finibus bibendum. Vivamus vehicula ligula vitae sodales venenatis. Cras blandit in ipsum eu bibendum. Morbi felis eros, pulvinar a bibendum sed, pellentesque vitae eros.\nMauris ut orci ut eros lobortis lobortis. Nam pellentesque nibh sed porta fringilla. Phasellus dapibus, erat tempus tincidunt pellentesque, felis sem eleifend mauris, vel rhoncus tellus massa et ante. Fusce faucibus ultrices ornare. Donec sed libero laoreet, auctor nulla ut, egestas felis. Duis quis consequat massa. Ut tincidunt semper mi. Praesent mattis ligula id ligula commodo venenatis. In ultrices elit vitae vehicula tincidunt. Ut consequat dignissim leo, ut lacinia risus eleifend id."
+
+// Bonus 1: word counter
+let wordCounter = 0;
+for (let i=0; i<paragraph.length; i++) {
+  if (paragraph[i] === " ") {
+    wordCounter++;
+  } else if (paragraph[i] === "." && paragraph[i+1] !== " ") {
+    wordCounter++;
+  }
+};
```

Comment:
os paso un ejemplo más sencillo con bucle for:

```
let wordsCount = 0;
for (x = 0; x < paragraph.length; x++) {
    if (paragraph[x] === ' ') {
        wordsCount++;
    }
};
```

y con métodos: 

`const wordCounter = paragraph.split(' ').length`